device/WakeOnLan/controller.js

```diff
@@ -0,0 +1,155 @@
+﻿'use strict';
+
+const BluePromise = require('bluebird');
+const fs = require('fs')
+const readline = require('readline')
+const os = require('os');
+const net = require('net');
+const dgram = require('dgram');
+
+const PORT_NUMBER = 9;
+const DEFAULT_IPADDRESS = '255.255.255.255';
+
+let wolMap = "wol.map";
+
+/**
+ * Sets the path to the 'wol.map' file
+ */
+module.exports.setMapPath = (mapPath) => {
+    wolMap = mapPath;
+}
+
+/**
+ * Handles button presses for WOL
+ */
+module.exports.onButtonPressed = (deviceid, name) => {
+    console.log(`WOL button pressed for ${name}`);
+
+    parseAddr(name)
+        .then((addr) => {
+            return Promise.all([addr, getMacBuffer(addr.macAddress)])
+        })
+        .then((results) => {
+            let magic = Buffer.alloc(6, 0xff);
+            for (let i = 0; i < 16; i++) {
+                magic = Buffer.concat([magic, results[1]]);
+            }
+            return Promise.all([results[0], magic]);
+        })
+        .then((results) => {
+            return sendPacket(results[0].ipAddress, results[1]);
+        })
+        .then((results) => {
+            console.log("WOL packet sent");
+        })
+        .catch((err) => {
+            console.error("Error occurred writing WOL packet. ", err || err.message);
+        });
+}
+
+/**
+ * Returns a promise to send the magic packet to the specified IP address
+ * @param {any} ipAddress the ipaddress to send to 
+ * @param {any} magic the magic packet to send
+ */
+function sendPacket(ipAddress, magic) {
+    return new BluePromise((resolve, reject) => {
+        const socket = dgram.createSocket(net.isIPv6(ipAddress) ? 'udp6' : 'udp4')
+            .on('error', (err) => {
+                socket.close();
+                reject(err);
+            })
+            .once('listening', () => {
+                socket.setBroadcast(true);
+            });
+
+        socket.send(magic, 0, magic.length, PORT_NUMBER, ipAddress, (err, res) => {
+            socket.close();
+            if (err === null || err === undefined) {
+                resolve(true);
+            } else {
+                reject("Exception writing the magic packet: " + err);
```

Comment:
reject should wrap the error message in a error object, like `reject(new Error('TEXT'))`, in your case you could just do a `reject(err)`.

to make it more readable you can

```
if (err) {
  reject(err);
} else {
  resolve();
}